pkg/dev_compiler/doc/GENERIC_METHODS.md

```diff
@@ -1,229 +1,164 @@
 
 ```dart
-// This declares a function which takes two unused generic method parameters
-int f/*<S, T>*/(int x) => 3;
+/// Takes two type parameters, [K] and [V].
+Map<K, V> singletonMap<K, V>(K key, V value) {
+  return <K, V>{ key, value };
+}
 ```
 
-As with classes, you can put bounds on type parameters.
+As with classes, you can put bounds on type parameters:
 
 ```dart
-// This declares a function which takes two unused generic method parameters
-// The first parameter (S) must extend num
-// The second parameter (T) must extend List<S>
-int f/*<S extends num, T extends List<S>>*/(int x) => 3;
+/// Takes a list of two numbers of some num-derived type [T].
+T sumPair<T extends num>(List<T> items) {
+  return items[0] + items[1];
+}
 ```
 
 Class methods (instance and static) can be declared to take generic parameters
-in the same way.
+in the same way:
 
 ```dart
 class C {
-  static int f/*<S, T>*/(int x) => 3;
-  int m/*<S, T>*/(int x) => 3;
+  static int f<S, T>(int x) => 3;
+  int m<S, T>(int x) => 3;
 }
 ```
 
-Function typed parameters, local functions, and function expressions can also be
-declared to take generic parameters.
+This even works for function-typed parameters, local functions, and function
+expressions:
 
 ```dart
-// foo takes a generic method as a parameter
-void foo(int f/*<S>*/(int x)) {}
+/// Takes a generic method as a parameter [callback].
+void functionTypedParameter(T callback<T>(T thing)) {}
 
-// foo declares a local generic function
-void foo() {
-  int f/*<S>*/(int x) => 3;
-  return;
+// Declares a local generic function `itself`.
+void localFunction() {
+  T itself<T>(T thing) => thing;
 }
 
-// foo binds a generic function expression to a local variable.
-void foo() {
-  var x = /*<S>*/(int x) => x;
+// Binds a generic function expression to a local variable.
+void functionExpression() {
+  var lambda = <T>(T thing) => thing;
 }
 ```
 
-We do not currently support a way to declare a function as returning a generic
-function.  This will eventually be supported using something analogous to Dart
-typedefs.
+We do not currently support a way to declare a function as *returning* a generic
+function. This will eventually be supported using a `typedef`.
 
-## Using generic method parameters
+## Using generic method type parameters
 
-The previous examples declared generic method parameters, but did not use them.
-You can use a generic method parameter `T` anywhere that a type is expected in
-Dart by writing a type followed by `/*=T*/`.  So for example, `dynamic /*=T*/`
-will be interpreted as `dynamic` by all non-strong mode tools, but will be
-interpreted as `T` by strong mode.  In places where it is valid to leave off a
-type, simply writing `/*=T*/` will be interpreted as `dynamic` by non-strong
-mode tools, but will be interpreted as `T` by strong mode.  For example:
+You've seen some examples already, but you can use a generic type parameter
+anywhere you would expect in a generic method.
```

Comment:
We should probably mention that it is currently an error to use it in an `is` check?